src/Fixer/PhpTag/ShortEchoTagFixer.php

```diff
@@ -0,0 +1,242 @@
+            $foundPrint = $tokens->findSequence([[\T_OPEN_TAG], [\T_PRINT]], $offset);
+            if (null === $foundEcho) {
+                $found = $foundPrint;
+                if (null === $found) {
+                    break;
+                }
+            } elseif (null === $foundPrint || key($foundEcho) < key($foundPrint)) {
+                $found = $foundEcho;
+            } else {
+                $found = $foundPrint;
+            }
+            $startRange = key($found);
+            $offset = $startRange + 1;
+            if (!$this->configuration[self::OPTION_SHORT_ALWAYS] && $this->isComplexCode($tokens, $offset)) {
+                continue;
+            }
+            next($found);
+            $endRange = key($found);
+            $newTokens = $this->buildLongToShortTokens($tokens, $startRange, $endRange);
+            $tokens->overrideRange($startRange, $endRange, $newTokens);
+        }
+    }
+
+    private function shortToLong(SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::LONG_FUNCTION_PRINT === $this->configuration[self::OPTION_LONG_FUNCTION]) {
+            $echoToken = [\T_PRINT, 'print'];
+        } else {
+            $echoToken = [\T_ECHO, 'echo'];
+        }
+        $offset = $tokens->count() - 1;
+        for (;;) {
+            $found = $tokens->getPrevTokenOfKind($offset, [[\T_OPEN_TAG_WITH_ECHO]]);
+            if (null === $found) {
+                break;
+            }
+            $offset = $found - 1;
+            $replace = [new Token([\T_OPEN_TAG, '<?php ']), new Token($echoToken)];
+            if (!$tokens[$found + 1]->isWhitespace()) {
+                $replace[] = new Token([\T_WHITESPACE, ' ']);
+            }
+            $tokens->overrideRange($found, $found, $replace);
+        }
+    }
+
+    private function isComplexCode(Tokens $tokens, $index)
+    {
+        $semicolonFound = false;
+        for ($count = $tokens->count(); $index < $count; ++$index) {
+            $token = $tokens[$index];
+            if ($token->isGivenKind(\T_CLOSE_TAG)) {
+                return false;
+            }
+            if (';' === $token->getContent()) {
+                $semicolonFound = true;
+            } elseif ($semicolonFound && !$token->isWhitespace()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Builds the list of tokens that replace a long echo sequence.
+     *
+     * @param \PhpCsFixer\Tokenizer\Tokens $tokens
+     * @param int                          $openTagIndex
+     * @param int                          $echoTagIndex
+     *
+     * @return \PhpCsFixer\Tokenizer\Token[]
+     */
+    private function buildLongToShortTokens(Tokens $tokens, $openTagIndex, $echoTagIndex)
+    {
+        $result = [new Token([\T_OPEN_TAG_WITH_ECHO, '<?='])];
+
+        // Skip to the first non-whitespace token between $openTagIndex and $echoTagIndex
```

Comment:
From description it looks like `Tokens::getNextNonWhitespace`.

Reply:
Not exactly the same: see the `between $openTagIndex and $echoTagIndex` part of the description

Reply:
Can we have a test for this "not exactly"? All tests are passing for https://github.com/mlocati/PHP-CS-Fixer/pull/1

Reply:
Yuu are right :+1:. That's because for sure the next non white-space token can only be an integer whose value is at max `$echoTagIndex`, it's better to use `getNextNonWhitespace()`